### PR TITLE
✅ Amp carousel flaky tests

### PR DESCRIPTION
**Pull request by krdwan**
Issue: https://github.com/ampproject/amphtml/issues/29783

### CODE REVIEW

File: test/integration/test-amp-carousel.js [hunk 1]
```diff
@@ -44,6 +46,11 @@ t.run('amp-carousel', function () {
       extensions,
     },
     (env) => {
+      async function waitForClass(el, className) {
+        const isButtonHidden = () => el.classList.contains(className);
```
Comment (samouri) [inline test/integration/test-amp-carousel.js:50]: nit: fn should be named: `hasClassCheck` or something like that.

File: test/integration/test-amp-carousel.js [hunk 2]
```diff
@@ -44,6 +46,11 @@ t.run('amp-carousel', function () {
...
+        await waitFor(isButtonHidden, 'button hidden');
```
Comment (samouri) [inline test/integration/test-amp-carousel.js:51]: nit: error message should be: `'did not receive class: ' + className`

File: test/integration/test-amp-carousel.js [hunk 3]
```diff
@@ -72,19 +79,33 @@ t.run('amp-carousel', function () {
...
           const nextBtn = amp.querySelector('.amp-carousel-button-next');
+
+          async function waitForLayout(el) {
+            const isButtonHidden = () =>
+              el.signals().get(CommonSignals.LOAD_END);
+            await waitFor(isButtonHidden, 'button hidden');
+          }
+
+          waitForLayout(amp);
+          waitForClass(prevBtn, 'amp-disable');
```
Comment (samouri) [inline test/integration/test-amp-carousel.js:98]: I think the right classname is `amp-disabled`

File: test/integration/test-amp-carousel.js [hunk 4]
```diff
@@ -72,19 +79,33 @@ t.run('amp-carousel', function () {
...
           const prevBtn = amp.querySelector('.amp-carousel-button-prev');
           const nextBtn = amp.querySelector('.amp-carousel-button-next');
+
+          async function waitForLayout(el) {
+            const isButtonHidden = () =>
+              el.signals().get(CommonSignals.LOAD_END);
+            await waitFor(isButtonHidden, 'button hidden');
+          }
+
+          waitForLayout(amp);
```
Comment (samouri) [inline test/integration/test-amp-carousel.js:97]: each of these `waitForLayout` and `waitForClass` needs to have an `await` before it

**Comment by krdwan** [comment]
Fixed in: https://github.com/ampproject/amphtml/pull/30933

**Pull request closed without merging**